### PR TITLE
[fix]: [CCM-12712]: Added docs annotations on BusinessMappingResource class

### PR DESCRIPTION
**Pull request by sdsingh07**
## Describe your changes
Added docs annotations on BusinessMappingResource class to view all the exposed APIs in the API docs.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/48784)

### CODE REVIEW

**Comment by sdsingh07** [comment]
trigger uts

**Review by iamanmolm — DISMISSED** [review]

File: ce-nextgen/service/src/main/java/io/harness/ccm/remote/resources/BusinessMappingResource.java [hunk 1]
```diff
@@ -82,10 +94,21 @@ public class BusinessMappingResource {
...
-  public RestResponse<BusinessMapping> save(
-      @QueryParam(NGCommonEntityConstants.ACCOUNT_KEY) String accountId, @Valid BusinessMapping businessMapping) {
+  @Operation(operationId = "createBusinessMapping",
+      description = "Create Cost category that allows you to categorize based on business requirements and "
+          + "get a contextual view of your expenses",
+      summary = "Create Cost category",
+      responses =
+      {
+        @io.swagger.v3.oas.annotations.responses.
+        ApiResponse(description = "Returns a created BusinessMapping object with all the cost and shared buckets",
```
Comment (iamanmolm) [inline ce-nextgen/service/src/main/java/io/harness/ccm/remote/resources/BusinessMappingResource.java:107]: Returns a created cost category object would make more sense here.

File: ce-nextgen/service/src/main/java/io/harness/ccm/remote/resources/BusinessMappingResource.java [hunk 2]
```diff
@@ -112,20 +145,42 @@ public RestResponse<BusinessMappingListDTO> list(@QueryParam(NGCommonEntityConst
...
   @ApiOperation(value = "Get Business Mapping", nickname = "getBusinessMapping")
-  public RestResponse<BusinessMapping> get(
-      @QueryParam(NGCommonEntityConstants.ACCOUNT_KEY) String accountId, @PathParam("id") String businessMappingId) {
+  @Operation(operationId = "getBusinessMapping",
+      description = "Fetch details of a Cost category for the given Cost category ID.",
+      summary = "Fetch details of a Cost category",
+      responses =
+      {
+        @io.swagger.v3.oas.annotations.responses.
+        ApiResponse(description = "Returns a BusinessMapping object with all the cost and shared buckets, "
```
Comment (iamanmolm) [inline ce-nextgen/service/src/main/java/io/harness/ccm/remote/resources/BusinessMappingResource.java:157]: Even here: "Returns a CostCategory object"

File: ce-nextgen/service/src/main/java/io/harness/ccm/remote/resources/BusinessMappingResource.java [hunk 3]
```diff
@@ -112,20 +145,42 @@ public RestResponse<BusinessMappingListDTO> list(@QueryParam(NGCommonEntityConst
...
-  public RestResponse<String> update(
-      @QueryParam(NGCommonEntityConstants.ACCOUNT_KEY) String accountId, BusinessMapping businessMapping) {
+  @Operation(operationId = "updateBusinessMapping",
+      description = "Update a Cost category. "
+          + "It accepts a BusinessMapping object and upserts it using the uuid mentioned in the definition.",
+      summary = "Update a Cost category",
+      responses =
+      {
+        @io.swagger.v3.oas.annotations.responses.
+        ApiResponse(description = "BusinessMapping CEView object with all the cost and shared buckets",
```
Comment (iamanmolm) [inline ce-nextgen/service/src/main/java/io/harness/ccm/remote/resources/BusinessMappingResource.java:179]: Here it returns just a message saying "Successfully updated the Business Mapping"

File: ce-nextgen/service/src/main/java/io/harness/ccm/remote/resources/BusinessMappingResource.java [hunk 4]
```diff
@@ -172,7 +239,7 @@ public RestResponse<CostCategoryDeleteDTO> delete(@NotEmpty @QueryParam(NGCommon
   @Path("linkedPerspectives")
   @Timed
   @Hidden
-  @Consumes(MediaType.APPLICATION_JSON)
+  @LogAccountIdentifier
   @ExceptionMetered
   @ApiOperation(value = "Get related Perspectives given Business Mapping Ids",
```
Comment (iamanmolm) [inline ce-nextgen/service/src/main/java/io/harness/ccm/remote/resources/BusinessMappingResource.java:244]: We can also change this to "given Cost Category Ids". And in the same function, everywhere we can call it cost category.

**Comment by sdsingh07** [comment]
trigger uts

**Comment by sdsingh07** [comment]
trigger messagecheck

**Review by iamanmolm — APPROVED** [review]

**Review by iamanmolm — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48784)* status: :shipit: complete! ✓

**Comment by sdsingh07** [comment]
trigger uts

**Comment by sdsingh07** [comment]
trigger ut0

**Pull request merged**